plugin/advancedsubscription/src/AdvancedSubscriptionPlugin.php

```diff
@@ -0,0 +1,872 @@
+                $where['AND session_id IN ( ? ) '] = implode($params['sessions']);
+            }
+            if (isset($params['status']) && is_array($params['status'])) {
+                $where['AND status IN ( ? ) '] = implode($params['status']);
+            }
+            $where['where'] = $where;
+            $count = Database::select('COUNT(*)', $advsubQueueTable, $where);
+            $count = $count[0]['COUNT(*)'];
+        }
+        return $count;
+    }
+
+    /**
+     * This method will call the Hook management insertHook to add Hook observer from this plugin
+     * @return int
+     */
+    public function installHook()
+    {
+        $hookObserver = HookAdvancedSubscription::create();
+        HookAdminBlock::create()->attach($hookObserver);
+        HookWSRegistration::create()->attach($hookObserver);
+        HookNotificationContent::create()->attach($hookObserver);
+        HookNotificationTitle::create()->attach($hookObserver);
+    }
+
+    /**
+     * This method will call the Hook management deleteHook to disable Hook observer from this plugin
+     * @return int
+     */
+    public function uninstallHook()
+    {
+        $hookObserver = HookAdvancedSubscription::create();
+        HookAdminBlock::create()->detach($hookObserver);
+        HookWSRegistration::create()->detach($hookObserver);
+        HookNotificationContent::create()->detach($hookObserver);
+        HookNotificationTitle::create()->detach($hookObserver);
+    }
+
+    /**
+     * Return the status from user in queue to session subscription
+     * @param int $userId
+     * @param int $sessionId
+     * @return bool|int
+     */
+    public function getQueueStatus($userId, $sessionId)
+    {
+        if (!empty($userId) && !empty($sessionId)) {
+            $queueTable = Database::get_main_table(TABLE_ADV_SUB_QUEUE);
+            $row = Database::select(
+                'status',
+                $queueTable,
+                array(
+                    'where' => array(
+                        'user_id = ? AND session_id = ?' => array($userId, $sessionId),
+                    )
+                )
+            );
+
+            if (count($row) == 1) {
+
+                return $row[0]['status'];
+            } else {
+
+                return ADV_SUB_QUEUE_STATUS_NO_QUEUE;
+            }
+        }
+
+        return false;
+    }
+
+    /**
+     * Return the remaining vacancy
+     * @param $sessionId
+     * @return bool|int
+     */
+    public function getVacancy($sessionId)
+    {
+        if (!empty($sessionId)) {
+            $extra = new ExtraFieldValue('session');
+            $var = $extra->get_values_by_handler_and_field_variable($sessionId, 'vacancies');
+            $vacancy = intval($var['field_value']);
+            if (!empty($vacancy)) {
+                $vacancy -= $this->countQueueByParams(array('sessions' => $sessionId, 'status' => ADV_SUB_QUEUE_STATUS_ADMIN_APPROVED));
```

Comment:
You mean the params + array should be spread on several lines? (always make sure you give a hint when referring to PSR, as we don't respect them all completely yet)